content/post/asahi-nixos/index.md

```diff
@@ -0,0 +1,100 @@
+---
+title: The Asahi NixOS Experience
+description: My 1.5 year experience with Asahi NixOS
+date: 2025-01-17 10:11:00 +0800
+categories:
+    - asahi-linux
+    - nix
+---
+
+In August of 2023, I was looking around for a new laptop. At the time,
+I was running an Alienware laptop with an RTX 3070 in it. However, it
+is a cumbersome machine with horrible battery life. The great thing was
+its performance. I needed a machine which was powerful, can run Linux,
+has great battery life, and portable. I considered x86 off the table as
+sleep modes on laptops has been a problem due to vendors not implementing
+them right. This led me to an ARM machine. Ultimately, I ended up with
+an Apple Macbook Pro from 2021 with an M1 Pro chip. It has 16GB of RAM and
+1TB of storage. To save money, I went with one from Apple's refurbishment
+service.
+
+## The Installation Process
+
+As soon as my Macbook Pro arrived, I began installing Nix on Darwin. It was
+a troublesome experience trying to integrate it properly in my Flake config.
+After a few hours, I was able to get home-manager working with Nix on Darwin.
+However, I wasn't done yet and I wanted Linux on this machine. So I stumbled
+upon [Thomas Watson](https://github.com/tpwrules)'s
+[`nixos-apple-silicon`](https://github.com/tpwrules/nixos-apple-silicon) Flake.
+
+The installation experience was quite smooth, all I needed was to run the Asahi
+Linux setup script in macOS after disabling the system integrity protection.
+Once I had SIP disabled and the Asahi Linux setup script executed, I then grabbed
+my USB flash drive with Thomas's Asahi NixOS installer on it. From there,
+installation proceeded like normal for NixOS. I set up an ext4 partition, pull my
+Flake, and run `nixos-install`.
+
+After some time of building, I then had NixOS on Apple Silicon. I had my
+desktop environment setup like how it is on my other machines but on ARM.
+Unfortunately, audio wasn't working properly at the time so I had to rely on
+Bluetooth. However, this became less of a problem after months had progressed.
+I've been happily running NixOS with Asahi packages on an M1 Pro.
+
+## The Problems
+
+A few problems I ran into with daily driving Asahi NixOS was the system
+did not have enough memory and not enough storage. I decided to get rid of
```

Comment:
I'm not sure there is a computer on earth with enough memory for your workload, Tristan.